src/routes/user.js

```diff
@@ -0,0 +1,101 @@
+import express from 'express';
+import prisma from '../tools/prisma.js';
+import { RequestError } from '../constants/commonErrors.js';
+import _ from 'lodash';
+
+const router = express.Router();
+
+router.get('/:id', async (request, response, next) => {
+  try {
+    const user = await prisma.user.findUnique({
+      where: {
+        id: request.params.id,
+      },
+    });
+    if (!user) {
+      throw new RequestError(`User not found with ID${request.params.id}`);
+    }
+    const userAttributes = _.pick(user, [
+      'name',
+      'email',
+      'createdAt',
+      'updatedAt',
+    ]);
+    response.json({ userAttributes });
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.patch('/:id', async (request, response, next) => {
+  try {
+    const user = await prisma.user.findUnique({
+      where: {
+        id: request.params.id,
+      },
+    });
+    if (!user) {
+      throw new RequestError(`User not found with ID${request.params.id}`);
+    }
```

Comment:
I'm looking at the commonError.js file and I'm seeing a lot of `Object.freeze(Error.prototype)`. Is that necessary for every single error? If so, what is the purpose of it?

Reply:
I see. When you mean "accidentally changed by anyone", are you referring to developers like us, or just people in general? Also couldn't I still just manually change the `this.statusCode` of, for example, AuthError from 403 to any number of I want? `Object.freeze()` wouldn't do anything to prevent that.

Another follow up question, what does the `.prototype` do in `Error.prototype` when you are passing in params to the freeze method?

Reply:
Makes sense! Thanks!